Flow.Launcher.Infrastructure/Alphabet.cs

```diff
@@ -19,160 +19,45 @@ public interface IAlphabet
-            
-            var pinyinArray=combination.Select(x => string.Join("", x));
-            var acronymArray = combination.Select(Acronym).Distinct();
-
-            var joinedSingleStringCombination = new StringBuilder();
-            var all = acronymArray.Concat(pinyinArray);
-            all.ToList().ForEach(x => joinedSingleStringCombination.Append(x));
-
-            return joinedSingleStringCombination.ToString();
-        }
-
-        public void Save()
-        {
-            if (!_settings.ShouldUsePinyin)
-            {
-                return; 
-            }
-
-            lock(_pinyinStorage)
-            {
-                _pinyinStorage.Save(PinyinCache.ToDictionary(i => i.Key, i => i.Value));
-            }            
         }
 
-        private static string[] EmptyStringArray = new string[0];
-        private static string[][] Empty2DStringArray = new string[0][];
 
-        /// <summmary>
-        /// replace chinese character with pinyin, non chinese character won't be modified
-        /// Because we don't have words dictionary, so we can only return all possiblie pinyin combination
-        /// e.g. 音乐 will return yinyue and yinle
-        /// <param name="characters"> should be word or sentence, instead of single character. e.g. 微软 </param>
-        /// </summmary>
-        public string[][] PinyinCombination(string characters)
+        public string Translate(string content)
         {
-            if (!_settings.ShouldUsePinyin || string.IsNullOrEmpty(characters))
-            {
-                return Empty2DStringArray;
-            }
-
-            if (!PinyinCache.ContainsKey(characters))
+            if (_settings.ShouldUsePinyin)
             {
-                var allPinyins = new List<string[]>();
-                foreach (var c in characters)
+                if (_pinyinCache.ContainsKey(content))
                 {
-                    var pinyins = PinyinHelper.toHanyuPinyinStringArray(c, Format);
-                    if (pinyins != null)
+                    if (WordsHelper.HasChinese(content))
                     {
-                        var r = pinyins.Distinct().ToArray();
-                        allPinyins.Add(r);
+                        var result = WordsHelper.GetPinyin(content, ";");
+                        result = GetFirstPinyinChar(result) + result.Replace(";", "");
+                        _pinyinCache[content] = result;
+                        return result;
                     }
                     else
                     {
-                        var r = new[] { c.ToString() };
-                        allPinyins.Add(r);
+                        return content;
                     }
                 }
-
-                var combination = allPinyins.Aggregate(Combination).Select(c => c.Split(';')).ToArray();
-                PinyinCache[characters] = combination;
-                return combination;
+                else
+                    return _pinyinCache[content];
```

Comment:
curly brackets please

Reply:
ok😉

Reply:
hey @taooceros, what @jjw24 meant here was to change to this:

```csharp
else
{
    return _pinyinCache[content];
}
```

Reply:
Sorry for misunderstanding, have edited.